src/compiler/program.ts

```diff
@@ -1168,9 +1168,6 @@ namespace ts {
                             let typeAssertionExpression = <TypeAssertion>node;
                             diagnostics.push(createDiagnosticForNode(typeAssertionExpression.type, Diagnostics.type_assertion_expressions_can_only_be_used_in_a_ts_file));
                             return true;
-                        case SyntaxKind.Decorator:

```

Comment:
since this is a non-standard feature, we should ask you to set a flag.. i.e. error if the flag is not set, and ask the user to specify the flag.